File: Doubao_MarsCode/java/MC0085/Main.java
package MC0085;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * 85. 数列差异的最小化
 * <p>
 * 运行超时
 */
public class Main {
    public static int solution(int n, int m, int k, List<Integer> a, List<Integer> b) {
        Collections.sort(a);
        Collections.sort(b);

        int minDiff = Integer.MAX_VALUE;
        int i = 0, j = 0;
        while (i < n && j < m) {
            int diff = Math.abs((a.get(i) - b.get(j)) * (a.get(i) - b.get(j)) - k * k);
            minDiff = Math.min(minDiff, diff);
            if (a.get(i) < b.get(j)) {
                i++;
            } else {
                j++;
            }
        }

        return minDiff;
    }

    public static void main(String[] args) {
        // You can add more test cases here
        // case 1
        List<Integer> a1 = Arrays.asList(5, 3, 4, 1, 2);
        List<Integer> b1 = Arrays.asList(0, 6, 7, 9, 8);
        System.out.println(solution(5, 5, 1, a1, b1) == 0);

        // case 2
        List<Integer> a2 = Arrays.asList(5, 3, 4, 1, 2);
        List<Integer> b2 = Arrays.asList(0, 6, 7, 9, 8);
        System.out.println(solution(5, 5, 0, a2, b2) == 1);

        // case 3
        List<Integer> a3 = Arrays.asList(5, 3, 4, 1, 2);
        List<Integer> b3 = Arrays.asList(0, 6, 7, 9, 8, 11);
        System.out.println(solution(5, 6, 3, a3, b3) == 0);
    }
}

File: Doubao_MarsCode/go/MC0085/main.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func solution(n int, m int, k int, a []int, b []int) int {
	sort.Ints(a)
	sort.Ints(b)

	minDiff := int(^uint(0) >> 1)
	i, j := 0, 0
	for i < n && j < m {
		diff := int(math.Abs(float64((a[i]-b[j])*(a[i]-b[j]) - k*k)))
		minDiff = min(minDiff, diff)
		if a[i] < b[j] {
			i++
		} else {
			j++
		}
	}

	return minDiff
}

func main() {
	// You can add more test cases here
	fmt.Println(solution(5, 5, 1, []int{5, 3, 4, 1, 2}, []int{0, 6, 7, 9, 8}) == 0)
	fmt.Println(solution(5, 5, 0, []int{5, 3, 4, 1, 2}, []int{0, 6, 7, 9, 8}) == 1)
	fmt.Println(solution(5, 6, 3, []int{5, 3, 4, 1, 2}, []int{0, 6, 7, 9, 8, 11}) == 0)
}
